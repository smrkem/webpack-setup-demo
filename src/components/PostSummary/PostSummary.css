.post-summary h2 {
  font-size: 3rem;
  color: #53384c;
}

.post-summary h2 a:focus,
.post-summary h2 a:hover {
  color: #990066;
  transition: color 0.3s ease;
}

.post-summary .post-excerpt,
.post-summary .post-content {
  font-size: 1.8rem;
  line-height: 1.7;
}
