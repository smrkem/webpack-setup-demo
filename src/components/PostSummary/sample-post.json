{
  "title": "Sample Post 2",
  "slug": "sample-post-2",
  "author": "Matt Smrke",
  "published_at": "2017-10-22",
  "excerpt": "Farm-to-table authentic heirloom chicharrones. Before they sold out pour-over quinoa normcore stumptown helvetica gastropub. Deep v succulents mustache synth cold-pressed. Drinking vinegar vaporware next level PBR&B pinterest la croix seitan. Tousled blue bottle iPhone glossier franzen jean shorts street art shaman stumptown next level. Fingerstache synth organic, single-origin coffee kombucha vinyl pinterest brooklyn you probably haven't heard of them tbh before they sold out hot chicken 3 wolf moon chicharrones meditation.",
  "content": "Farm-to-table authentic heirloom chicharrones. Before they sold out pour-over quinoa normcore stumptown helvetica gastropub. Deep v succulents mustache synth cold-pressed. Drinking vinegar vaporware next level PBR&B pinterest la croix seitan. Tousled blue bottle iPhone glossier franzen jean shorts street art shaman stumptown next level. Fingerstache synth organic, single-origin coffee kombucha vinyl pinterest brooklyn you probably haven't heard of them tbh before they sold out hot chicken 3 wolf moon chicharrones meditation. Bespoke activated charcoal you probably haven't heard of them pop-up drinking vinegar jean shorts tumblr occupy. Typewriter chicharrones tumeric heirloom hella, tousled sartorial polaroid 90's vegan neutra trust fund twee tattooed keffiyeh. Live-edge air plant hammock polaroid four dollar toast dreamcatcher kogi sriracha vexillologist pug.\n\nLeggings whatever vegan palo santo hoodie tote bag. Marfa lyft typewriter tofu hashtag mumblecore tumblr trust fund jean shorts migas. Tofu YOLO keytar messenger bag stumptown pabst selfies kombucha. Af shaman 8-bit chartreuse sriracha irony godard live-edge cred. Poke roof party activated charcoal forage. Live-edge actually umami leggings tousled. Bitters la croix ethical microdosing, air plant actually fixie ennui. Pour-over cardigan etsy four dollar toast, glossier cornhole flannel copper mug 3 wolf moon hammock pabst hot chicken keytar seitan. Taiyaki pok pok man bun street art intelligentsia biodiesel. Man bun locavore air plant hot chicken mixtape selvage ramps. Lomo marfa jianbing intelligentsia vice letterpress, tofu artisan retro tattooed.\n\nEtsy bitters DIY farm-to-table try-hard, jianbing waistcoat everyday carry leggings. Shabby chic cliche lo-fi literally green juice, live-edge street art letterpress enamel pin unicorn. Migas literally jean shorts, beard tumeric succulents edison bulb adaptogen deep v fixie twee fam chia keffiyeh intelligentsia. Cardigan everyday carry green juice jean shorts helvetica mustache lyft man braid enamel pin ramps polaroid venmo bespoke lomo hella. Stumptown crucifix brunch flexitarian shaman organic, etsy glossier slow-carb put a bird on it master cleanse blog seitan actually. Bespoke neutra scenester, glossier food truck authentic seitan. Craft beer artisan man braid viral. Vape selfies marfa lumbersexual mixtape celiac disrupt VHS selvage squid before they sold out farm-to-table blog. Coloring book bitters fashion axe semiotics typewriter banh mi chia cloud bread hella literally locavore vape fingerstache bespoke. Cornhole pinterest chicharrones, wolf helvetica kinfolk fam williamsburg poutine organic mlkshk iPhone franzen poke cold-pressed. Pug authentic vexillologist, squid brunch vaporware air plant fanny pack fixie dreamcatcher hot chicken tofu blog banh mi."
}
