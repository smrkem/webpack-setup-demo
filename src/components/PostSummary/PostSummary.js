import React from 'react'
import { Link } from 'react-router-dom'

import './PostSummary.css'

const PostSummary = (props) => (
  <div className="post-summary">
    <h2>
      <Link to={`posts/${props.slug}`}>
        { props.title }
      </Link>
    </h2>
    <div className="post-excerpt">
      { props.excerpt }
    </div>
  </div>
)

export default PostSummary
