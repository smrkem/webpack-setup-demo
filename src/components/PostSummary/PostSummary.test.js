import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PostSummary from './PostSummary'

import post from './sample-post.json'

test('PostSummary should render ok', () => {
  const component = shallow(<PostSummary {...post} />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
