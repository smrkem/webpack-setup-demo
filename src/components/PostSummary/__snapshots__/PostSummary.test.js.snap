// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSummary should render ok 1`] = `
<div
  className="post-summary"
>
  <h2>
    <Link
      replace={false}
      to="posts/sample-post-2"
    >
      Sample Post 2
    </Link>
  </h2>
  <div
    className="post-excerpt"
  >
    Farm-to-table authentic heirloom chicharrones. Before they sold out pour-over quinoa normcore stumptown helvetica gastropub. Deep v succulents mustache synth cold-pressed. Drinking vinegar vaporware next level PBR&B pinterest la croix seitan. Tousled blue bottle iPhone glossier franzen jean shorts street art shaman stumptown next level. Fingerstache synth organic, single-origin coffee kombucha vinyl pinterest brooklyn you probably haven't heard of them tbh before they sold out hot chicken 3 wolf moon chicharrones meditation.
  </div>
</div>
`;
